qermit/noise_model/transpiler_backend.py

```diff
@@ -0,0 +1,186 @@
+        max_batch_size: Shots are simulated in batches. This is the largest
+            shot batch size permitted.
+        result_dict: A dictionary mapping handles to results.
+        backend: Backend used to simulate compiled circuits.
+    """
+
+    transpiler: BasePass
+    max_batch_size: int
+    result_dict: Dict[uuid.UUID, BackendResult]
+    backend = AerBackend()
+
+    def __init__(
+        self,
+        transpiler: BasePass,
+        max_batch_size: int = 100,
+        result_dict: Dict[uuid.UUID, BackendResult] = {},
+    ):
+        """Initialisation method.
+
+        :param transpiler: Compiler to use during noise simulation.
+        :type transpiler: BasePass
+        :param max_batch_size: Size of the largest batch of shots,
+            defaults to 100
+        :type max_batch_size: int, optional
+        :param result_dict: Results dictionary, may be used to store existing
+            results within backend, defaults to {}
+        :type result_dict: Dict[uuid.UUID, BackendResult], optional
+        """
+
+        self.transpiler = transpiler
+
+        self.max_batch_size = max_batch_size
+        self.result_dict = result_dict
+
+    def run_circuit(
+        self,
+        circuit: Circuit,
+        n_shots: int,
+        **kwargs,
+    ) -> BackendResult:
+        """Return results of running one circuit.
+
+        :param circuit: Circuit to run
+        :type circuit: Circuit
+        :param n_shots: Number of shots to be taken from circuit.
+        :type n_shots: int
+        :return: Result of running circuit.
+        :rtype: BackendResult
+        """
+
+        handle = self.process_circuit(circuit, n_shots, **kwargs)
+        return self.get_result(handle=handle)
+
+    def process_circuit(
+        self,
+        circuit: Circuit,
+        n_shots: int,
+        **kwargs,
+    ) -> uuid.UUID:
+        """[summary]
+
+        :param circuit: Submits circuit to run on noisy backend.
+        :type circuit: Circuit
+        :param n_shots: Number of shots to take from circuit.
+        :type n_shots: int
+        :return: Handle identifying results in `result_dict`.
+        :rtype: uuid.UUID
+        """
+
+        handle = uuid.uuid4()
+
+        counts = self.get_counts(
+            circuit=circuit,
+            n_shots=n_shots,
+            cbits=circuit.bits,
+        )
+
+        self.result_dict[handle] = BackendResult(
+            counts=Counter({
+                OutcomeArray.from_readouts([key]): val
+                for key, val in counts.items()
+            }),
+            c_bits=circuit.bits,
+        )
+
+        return handle
+
+    def get_result(self, handle: uuid.UUID) -> BackendResult:
+        """Retrive result from backend.
+
+        :param handle: Handle identifying result.
+        :type handle: uuid.UUID
+        :return: Result corresponding to handle.
+        :rtype: BackendResult
+        """
+        return self.result_dict[handle]
+
+    def get_counts(
```

Comment:
Looks like something weird has happened with the formatting in this file? the `get_counts` method has `gen_transpiled_circuit` inside its definition?

Reply:
It does indeed, and that's intentional. I don't want to expose gen_transpiled_circuit as a class method as the circuit generated is not of use outside of the get counts method. If the user wants to generate a noisy circuit themselves they should use the transpiiler.

Reply:
I'd reccomend moving `gen_transpiled_circuit` outside of this function definition, but renaming it to `_gen_transpiled_circuit`, which is standard practice for signifying a class method is for internal use instead of external use